lib/manager/npm/post-update/yarn.ts

```diff
@@ -46,15 +49,38 @@ export async function generateLockFile(
   logger.debug(`Spawning yarn install to create ${lockFileName}`);
   let lockFile = null;
   try {
-    let installYarn = 'npm i -g yarn';
-    const yarnCompatibility = config.compatibility?.yarn;
-    if (validRange(yarnCompatibility)) {
-      installYarn += `@${quote(yarnCompatibility)}`;
+    const configuration = await Configuration.find(
+      npath.toPortablePath(cwd),
+      null,
+      { strict: false }
+    ).catch(() => null);
+    const yarnPath = configuration && configuration.get('yarnPath');
```

Comment:
it seems the path is a fullpath, would that be a problem with exec ? it seems exec is done in docker ?

Reply:
We should move the version extraction to the extract phase and use config.compatibility.yarn here

Reply:
to clarify: yarnPath contains yarn 2 executable path witch is in the repo

Reply:
I know, I mean to fetch the yarn version, which should be done at extraction phase.